bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/RevogiSmartStripControlHandler.java

```diff
@@ -0,0 +1,172 @@
+import org.openhab.binding.revogi.internal.api.SwitchService;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RevogiSmartStripControlHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Andi Bräu - Initial contribution
+ */
+@NonNullByDefault
+public class RevogiSmartStripControlHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(RevogiSmartStripControlHandler.class);
+    private final StatusService statusService;
+    private final SwitchService switchService;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private @Nullable RevogiSmartStripControlConfiguration config;
+
+    public RevogiSmartStripControlHandler(Thing thing, StatusService statusService, SwitchService switchService) {
+        super(thing);
+        this.statusService = statusService;
+        this.switchService = switchService;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case RevogiSmartStripControlBindingConstants.PLUG_1_SWITCH:
+                switchPlug(command, 1);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_2_SWITCH:
+                switchPlug(command, 2);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_3_SWITCH:
+                switchPlug(command, 3);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_4_SWITCH:
+                switchPlug(command, 4);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_5_SWITCH:
+                switchPlug(command, 5);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_6_SWITCH:
+                switchPlug(command, 6);
+                break;
+            case RevogiSmartStripControlBindingConstants.ALL_PLUGS:
+                switchPlug(command, 0);
+                break;
+            default:
+                logger.info("Sometring went wrong, we've got a message for {}", channelUID.getId());
+        }
+    }
+
+    private void switchPlug(Command command, int port) {
+        if (config == null) {
+            logger.warn("No config available, config object was null");
+            return;
+        }
+        if (command instanceof OnOffType) {
+            int state = convertOnOffTypeToState(command);
+            switchService.switchPort(config.getSerialNumber(), config.ipAddress, port, state);
```

Comment:
Since `config` is Nullable, you need to store it to a local variable and check if it is null before using it. Same for below. Please check all compiler warnings.